src/common/Thread.cc

```diff
@@ -22,6 +22,13 @@
 #include <errno.h>
 #include <iostream>
 #include <pthread.h>
+#if defined(__FreeBSD__)
+// pthread non-POSIX stuff is in a separate include file
```

Comment:
minor, i think this condition macro is self-explained, so no need to have these comments?